sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java

```diff
@@ -1283,6 +1283,10 @@ private GloballyAsSingletonView(
               insertDefault,
               insertDefault ? fn.defaultValue() : null,
           combined.getCoder());
+
+      // GBK's inside view writers for singleton view should discard previous values.
+      materializationInput.setWindowingStrategyInternal(input.getWindowingStrategy()
```

Comment:
You should instead use `Window.configure().discardingFiredPanes()`, explicitly setting any other required values from the upstream `WindowingStrategy`

Reply:
`Window.configure()` does not seem to work. I will take a little break from this bug and re-visit it soon.